src/CodeGenerator/ClassBuilder.cs

```diff
@@ -0,0 +1,301 @@
+﻿using Antlr4.StringTemplate;
+using CodeGenerator.Templates;
+using CodeGenerator.Templates.Resources;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace CodeGenerator
+{
+    public class ClassBuilder
```

Comment:
Not sold on encapsulating with a builder vs encapsulating with a class (`Class`, e.g. how `Method` was done)

Keep as-is but revisit once this starts seeing real use